src/bin/lpython.cpp

```diff
@@ -269,55 +267,84 @@ int emit_llvm(const std::string &infile,
     return 0;
 }
 
+void print_report(std::vector <std::pair<std::string, double>> &times, bool allow) {
+    if (!allow || times.size() == 0)
+        return;
+    for (std::pair<std::string, double> &stage: times) {
+        std::cout << stage.first << ": " << stage.second << "ms" << std::endl;
+    }
+}
+
 int compile_python_to_object_file(
         const std::string &infile,
         const std::string &outfile,
         const std::string &runtime_library_dir,
-        CompilerOptions &compiler_options)
-{
-    Allocator al(4*1024);
+        const bool time_report,
+        CompilerOptions &compiler_options) {
+
+    Allocator al(4 * 1024);
```

Comment:
If you want to make formatting changes, you can, but please send them as a separate PR. In a PR where you are implementing new functionality, please do not change other code by adding white spaces, as it makes it harder to review, and also it seems to me these formatting changes are not necessarily making it more readable, but we should have that discussion on a separate PR, so that we can focus here on the actual functionality.